### ISSUE TITLE
Environment variables not correctly resolved in server.xml variables

### ISSUE THREAD

**Issue by tevans78** [OpenLiberty/open-liberty#8298]
Variables in the server.xml are exposed through MP Config. e.g.

```
@Inject @ConfigProperty("javaHome")
String javaHome;
```

It should be possible to use environment variables in the value defined in server.xml. All of the following should resolve the same...

`<variable name="javaHome" value="${java.home}"/>`
`<variable name="javaHome" value="${java_home}"/>`
`<variable name="javaHome" value="${JAVA_HOME}"/>`
`<variable name="javaHome" value="${env.JAVA_HOME}"/>`

None of them work properly. The last one (with an "env." prefix) can be made to work by adding an extra call to `VariableRegistry.resolveString(String)` in the MP Config code.

**Comment by bpaskin-ibm** [comment]
This goes beyond the just system environment variables, but includes variables that receive their value from the `bootstrap.properties` file.  The value is properly propagated to the server.xml, but when mpConfig is used the injected variable does not have the value from the `bootstrap.properties`, but the configuration value from server.xml (app -> server.xml -> bootstrap.properties). 

Brian

**Comment by brenthdaniel** [comment]
Config is providing precisely what was requested by MP through the ConfigVariables interface (namely, only variables defined in server.xml.) 

We never exposed ConfigVariables as SPI, so it's fine to change it if you don't want to use the VariableRegistry SPI to resolve them, but is there any potential impact to existing applications of providing extra information in this config source?

**Comment by tevans78** [comment]
I think the ConfigVariables SPI probably should change behaviour, to resolve any of the 4 patterns mentioned above. I am not aware of any applications/customers who might be relying on the existing behaviour but that doesn't mean there aren't any. This change would probably have to go to the POC? @Emily-Jiang what do you think? What would you like us to do?

**Comment by Emily-Jiang** [comment]
I would like the resolved values for the 4 cases you listed. Not resolving the variables is half-baked. If this need to got to PoC, we should. By the way, @brenthdaniel can you create another method to resolve the variables, so that MP Config can use that?

**Comment by brenthdaniel** [comment]
I don't think we need to create a new method -- Either we should change the existing interface to return resolved variables or MP should use the VariableRegistry interface to resolve them. 

My guess is that there would be little customer impact from this, but it's a theoretical possibility.

**Comment by tevans78** [comment]
I think the best option is to change the existing interface ... but, when I tried using the VariableRegistry interface, it only worked for the `env.` version, it didn't find the other three.

**Comment by sdaschner** [comment]
My 2ct on the current behavior: What is really confusing is that in the documentation it's described that the `server.xml` is capable of substituting such values (coming from environment variables, for example), which sounds like you could simply use the cases described above. From a naïve user's perspective, I'd expect that to work in this way (or it should be documented further). The easiest way for a user is if these values would work, regardless in which element / context, if that makes sense.

**Comment by TJManschot** [comment]
I'm currently on 23.0.0.6 and I still run into this issue. Is there a workaround that allows me to use environment variables in the server.xml for mp-config?
